drivers/sensor/am2320/am2320.c

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2017 Tidy Jiang <tidyjiang@163.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <i2c.h>
+#include <init.h>
+#include <misc/__assert.h>
+#include <misc/byteorder.h>
+#include <sensor.h>
+#include <string.h>
+#include <crc16.h>
+#include <logging/sys_log.h>
+
+/* AM2320 I2C Address, 7 bits */
+#define AM2320_I2C_ADDR         0x5C
+
+/**
+ * Note: In order to wake up AM2320, we need to send START +
+ * ADDRESS(0xB8) signal to it. But I can't find a way to send
+ * this signal, so I use API i2c_write() with a arbitrary
+ * value (here is 0xB8) to do so.
+ */
+#define AM2320_WAKEUP_VALUE     0xB8
+
+/* AM2320 Function Code, Read/Write */
+#define AM2320_FUNCCODE_READ    0x03
+#define AM2320_FUNCCODE_WRITE   0X10
+
+/* Register's Start address and Length to fetch data. */
+#define AM2320_FETCH_START_ADDR 0X00
+#define AM2320_FETCH_LEN        0X04
+
+struct am2320_data {
+	struct device *i2c;
+	s16_t temp;
+	s16_t humidity;
+};
+
+static u16_t am2320_crc16(u8_t *ptr, unsigned int len)
+{
+	u16_t   crc = 0xFFFF;
+	u8_t    i;
+
+	while (len--) {
+		crc ^= *ptr++;
+
+		for (i = 0; i < 8; i++) {
+			if (crc & 0x01) {
+				crc >>= 1;
+				crc ^= 0xA001;
+			} else {
+				crc >>= 1;
+			}
+		}
+	}
+
+	return crc;
+}
+
+static int am2320_channel_get(struct device *dev,
+				enum sensor_channel chan,
+				struct sensor_value *val)
+{
+	struct am2320_data *drv_data = dev->driver_data;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_TEMP || SENSOR_CHAN_HUMIDITY);
+
+	if (chan == SENSOR_CHAN_TEMP) {
+		val->val1 = drv_data->temp / 10;
```

Comment:
Please add a comment that describes what units you are converting from and to. Same for humidity